### DISCUSSION TITLE
How can I convert my jpeg image stored in protobuf to foxglove.CompressedImage

### DISCUSSION THREAD

**Discussion by dvorak0** [foxglove/community#628] [Q&A]
I use mcap writer to convert my image(serialized by protobuf) to mcap files. The image protobuf is like
``` protobuf
message ImageData {
  Image image = 1;
}

message Image {
  oneof image_type {
    bytes encoded_str = 1;
    RgbImage rgb_image = 2;
  }
}
``` 
my convertor code is like:
``` typescript
import { ExtensionContext } from "@foxglove/studio";

type Image = {
    encoded_str: Uint8Array;
};

type MyImage = {
    image: Image;
};

export function activate(extensionContext: ExtensionContext) {
    extensionContext.registerMessageConverter({
        fromSchemaName: "ImageData",
        toSchemaName: "foxglove.CompressedImage",
        converter: (myImage: MyImage) => {
            return {
                data: myImage.image.encoded_str,
                format: "jpeg",
            };
        },
    });
}
```

But I can not select my image topic in Image panel.

help needed. Thanks!

Or where I can find helpful logging regarding extensions?

**Comment by defunctzombie** [comment]
We are currently working on adding support for message converter extensions to the image panel. Unfortunately right now it does not support them.

**Reply by zhenfeiyang** [reply]
Could I know if the map panel (gps message) is the only panel with converter support?

**Reply by defunctzombie** [reply]
The 3d panel has converter support as well.

**Reply by jtbandes** [reply]
You can follow this PR for image panel support: https://github.com/foxglove/studio/pull/5332

**Reply by jhurliman** [reply]
https://github.com/foxglove/studio/pull/5332 is merged, the next release of desktop Foxglove Studio will support message converters in the image panel.
